src/core/services/theming/theming.js

```diff
@@ -153,6 +161,9 @@ function ThemingProvider($mdColorPalette) {
     extendPalette: extendPalette,
     theme: registerTheme,
 
+    registerStyles: function(styles) {
```

Comment:
Could you clarify what you mean by compile? I would expect the provided styles are valid CSS just like $MD_THEME_CSS and will just be appended to the constant before the themes are generated.